contracts/specs/NFTSwapper/NFTSwapperSpec.md

```diff
@@ -0,0 +1,163 @@
+
+Requirements:
+* This contract must be approved to transfer the `nft` with `inputTokenId` on behalf of `msg.sender`.
+* This contract must be approved to transfer the `nft` with `outputTokenId` on behalf of the `swapPool`
+* Swap fee of `swapPrice(inputTokenId, outputTokenId)` Wei must be provided 
+
+
+## Swap Price Curve 
+
+The swap price is determined by the curve defined below. The `PriceCurveParams` are configured on deployment and cannot be changed.
+
+> Currently 3 different options are being considered, only 1 will be selected once finalized
+> - [Constant Price Curve](#constant-price-curve)
+> - [Output NFT age based curve](#output-nft-age-based-curve)
+> - [Input and Output NFT age difference based curve](#input-and-output-nft-age-difference-based-curve)
+
+### Constant Price Curve
+
+Simplest price curve where every swap is the same `floor` price.
+
+```solidity
+struct PriceCurveParams {
+    uint256 floor;
+}
+
+PriceCurveParams immutable priceCurveParams;
+
+function swapPrice() pure returns (uint256) {
+    return priceCurveParams.floor;
+}
+```
+
+![](img/constant-price-curve.png)
+
+Pros:
+* Simplicity
+
+Cons:
+* Values all NFTs equally, worst for maintaining non-fungibility.
+
+### Output NFT age based curve
+
+Prices lower `outputTokenId` at a higher value. 
+
+```solidity
+struct PriceCurveParams{
+    uint256 floor;
+    uint256 slope;
+}
+
+PriceCurveParams immutable priceCurveParams;
+
+function swapPrice(uint256 outputTokenId) view returns (uint256) {
+    uint256 outputAge = nft.totalSupply() - outputTokenId;
+    return priceCurveParams.floor + outputAge * priceCurveParams.slope;
+}
+```
+
+![](img/output-age-based-price-curve.png)
+
+Pros:
+* Values output NFTs distinctly, where lower ID's are more expensive to swap for.
+
+Cons: 
+* Treats all `inputTokenIds` as being valued equally (i.e treats as fungible)
+* Requires the ERC721 is `ERC721Enumerable` to support the `totalSupply()` 
+* Time variance, the `swapPrice` gets more expensive every day (could argue this is a pro)
+
+### Input and Output NFT age difference based curve
+
+Price based on the difference between `inputTokenId` and `outputTokenId` (or age difference). 
+
+```solidity
+struct PriceCurveParams{
+    uint256 floor;
+    uint256 slope;
+}
+
+PriceCurveParams immutable priceCurveParams;
+
+function swapPrice(uint256 inputTokenId, uint256 outputTokenId) pure returns (uint256) {
+    (, uint256 ageDiff) = inputTokenId.trySub(outputTokenId); // Clamps to 0
+    return priceCurveParams.floor + ageDiff * priceCurveParams.slope;
+}
+```
+
+![](img/age-difference-price-curve.png)
+
+Pros:
+* Values output NFTs distinctly, where lower ID's are more expensive to swap for.
+* Values input NFTs distinctly, where lower ID's are less expensive to swap from.
+* Time invariance, the `swapPrice` today will be the same as tomorrow for the same swap.
+
+Cons:
+* Most complex in terms of user understanding (?) 
```

Comment:
It's most complex but is also pretty straightforward. I also like how the swap price is the same no matter when you swap. If we are going to value ids then the id you are giving should be valued as well.

Reply:
Yep, agreed here, I lean toward this price curve